app/src/main/java/com/oracle/javafx/scenebuilder/app/welcomedialog/WelcomeDialogFilesDropHandler.java

```diff
@@ -0,0 +1,128 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package com.oracle.javafx.scenebuilder.app.welcomedialog;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class WelcomeDialogFilesDropHandler {
+
+    private static final Logger LOGGER = Logger.getLogger(WelcomeDialogFilesDropHandler.class.getName());
+
+    private final List<File> droppedFiles;
+    private final List<String> toOpen;
+    private final List<String> unsupportedItems;
+    private Consumer<List<String>> openFiles;
+    private Consumer<List<String>> handleUnsupported;
+
+    WelcomeDialogFilesDropHandler(List<File> droppedFiles) {
+        this.droppedFiles = Objects.requireNonNull(droppedFiles);
+        this.toOpen = new ArrayList<>(droppedFiles.size());
+        this.unsupportedItems = new ArrayList<>(droppedFiles.size());
+    }
+
+    final WelcomeDialogFilesDropHandler withSupportedFiles(Consumer<List<String>> handleOpen) {
+        this.openFiles = handleOpen;
+        return this;
+    }
+
+    final WelcomeDialogFilesDropHandler withUnsupportedFiles(Consumer<List<String>> unsupportedHandler) {
+        this.handleUnsupported = unsupportedHandler;
+        return this;
+    }
+
+    final void run() {
+        analyzeDroppedItems();
+        handleDropResult();
+    }
+
+    final void handleDropResult() {
+        if (this.openFiles == null) {
+            throw new IllegalStateException("Please configure a dropped file handling action using the withSupportedFiles(...) method.");
+        }
+        if (this.handleUnsupported == null) {
+            throw new IllegalStateException("Please configure an action for handling of unsupported files using withUnsupportedFiles(...) method.");
+        }
+        
+        if (!toOpen.isEmpty()) {
+            LOGGER.log(Level.INFO, "Received drop event to open files...");
+            openFiles.accept(toOpen);
+        } else {
+            LOGGER.log(Level.INFO, "Dropped object does not contain any loadable FXML files.");
+            handleUnsupported.accept(unsupportedItems);
+        }
+    }
+
+    final void analyzeDroppedItems() {
+        if (droppedFiles.isEmpty()) {
+            return;
+        }
+
+        for (var file : droppedFiles) {
+            if (file.isDirectory()) {
+                File[] children = file.listFiles();
+                List<String> inDir = new ArrayList<>(children.length);
+                for (var child : children) {
+                    if (isFxml(child)) {
+                        inDir.add(child.getAbsolutePath());
+                    }
+                }
+                if (inDir.isEmpty()) {
+                    unsupportedItems.add(file.getAbsolutePath());
+                } else {
+                    toOpen.addAll(inDir);
+                }
```

Comment:
In case we add multiple items to SB and one of them is valid, it won't show the dialog box for others. Is this the desired behaviour?

Reply:
Good catch. I'll look again into this and will come up with an alternative proposal. As of now it seems not to be the desired behavior.

Reply:
This is not desired. Its indeed incorrect. I'll rework this. Eventually there will be also a message for unsupported elements. However, the question is, is this of advantage from user perpective?

My idea was to open what is supported. FXML files and may be folders with FXMLs. When dropping empty folders or images? Why bother? Whats the better behavior?

Reply:
I prefer the previous implementation. If we want to be silent let's be silent when there is a valid file :)

Reply:
I agree. so I'll revert and rethink.

Reply:
Its now reverted and I added the conceptual explanation as Javadoc. There is more detailed logging for the case.

Reply:
Sounds good. Can this PR be moved to "Ready to review" now?

Reply:
Its in review now.